pubsub.go

```diff
@@ -326,11 +327,7 @@ type MsgIdFunction func(pmsg *pb.Message) string
 // but it can be customized to e.g. the hash of the message.
 func WithMessageIdFn(fn MsgIdFunction) Option {
 	return func(p *PubSub) error {
-		p.msgID = fn
-		// the tracer Option may already be set. Update its message ID function to make options order-independent.
-		if p.tracer != nil {
-			p.tracer.msgID = fn
-		}
```

Comment:
Not relevant anymore as now we ref to `DefaultMsgIdFunc` in `msgIdGenerator`

Reply:
:+1: